File: db4oj.tests/src/com/db4o/db4ounit/common/types/arrays/SimpleStringArrayTestCase.java
/* Copyright (C) 2006  Versant Inc.  http://www.db4o.com */

package com.db4o.db4ounit.common.types.arrays;

import db4ounit.*;
import db4ounit.extensions.*;

public class SimpleStringArrayTestCase extends AbstractDb4oTestCase {

	private static final String[] ARRAY = new String[] {"hi", "babe"};

	public static class Data {
	    public String[] _arr;

		public Data(String[] _arr) {
			this._arr = _arr;
		}
	}

	protected void store() throws Exception {
		db().store(new Data(ARRAY));
	}
	
    public void testRetrieve(){
    	Data data=(Data) retrieveOnlyInstance(Data.class);
    	ArrayAssert.areEqual(ARRAY,data._arr);
    }
}

File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/Types/Arrays/SimpleStringArrayTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4oUnit;
using Db4oUnit.Extensions;
using Db4objects.Db4o.Tests.Common.Types.Arrays;

namespace Db4objects.Db4o.Tests.Common.Types.Arrays
{
	public class SimpleStringArrayTestCase : AbstractDb4oTestCase
	{
		private static readonly string[] Array = new string[] { "hi", "babe" };

		public class Data
		{
			public string[] _arr;

			public Data(string[] _arr)
			{
				this._arr = _arr;
			}
		}

		/// <exception cref="System.Exception"></exception>
		protected override void Store()
		{
			Db().Store(new SimpleStringArrayTestCase.Data(Array));
		}

		public virtual void TestRetrieve()
		{
			SimpleStringArrayTestCase.Data data = (SimpleStringArrayTestCase.Data)((SimpleStringArrayTestCase.Data
				)RetrieveOnlyInstance(typeof(SimpleStringArrayTestCase.Data)));
			ArrayAssert.AreEqual(Array, data._arr);
		}
	}
}
